packages/api/tests/endpoints/test-rules.js

```diff
@@ -257,9 +300,59 @@ test('403 error when calling the API endpoint to delete an existing rule without
 });
 
 test('POST creates a rule', async (t) => {
-  const newRule = fakeRuleFactoryV2({
-    name: 'make_waffles',
+  const { dbClient } = t.context;
+
+  const newRule = fakeRuleFactoryV2(
+    {
+      executionNamePrefix: 'somePrefix',
+      queueUrl: 'queue_url',
+      meta: {},
+      tags: ['tag'],
+    }
+  );
+
+  const fakeCollection = fakeCollectionFactory();
+  const fakeProvider = fakeProviderFactory({
+    encrypted: true,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+    createdAt: new Date(2020, 11, 17),
+    updatedAt: new Date(2020, 12, 2),
   });
+
+  newRule.provider = fakeProvider.id;
+  newRule.collection = {
+    name: fakeCollection.name,
+    version: fakeCollection.version,
+  };
+
+  const collectionRecord = {
+    name: fakeCollection.name,
+    version: fakeCollection.version,
+    duplicate_handling: fakeCollection.duplicateHandling,
+    granule_id_validation_regex: fakeCollection.granuleId,
+    granule_id_extraction_regex: fakeCollection.granuleIdExtraction,
+    files: (JSON.stringify(fakeCollection.files)),
+    report_to_ems: fakeCollection.reportToEms,
+    sample_file_name: fakeCollection.sampleFileName,
+    created_at: new Date(fakeCollection.createdAt),
+    updated_at: new Date(fakeCollection.updatedAt),
+  };
+  const providerRecord = {
+    created_at: fakeProvider.createdAt,
+    updated_at: fakeProvider.updatedAt,
+    name: fakeProvider.id,
+    cm_key_id: fakeProvider.cmKeyId,
+    certificate_uri: fakeProvider.certificateUri,
+    private_key: fakeProvider.privateKey,
+    host: fakeProvider.host,
+    port: fakeProvider.port,
+  };
+
+  const [collectionCumulusId = {}] = await dbClient(tableNames.collections).insert(collectionRecord).returning('cumulus_id');
+  const [providerCumulusId = {}] = await dbClient(tableNames.providers).insert(providerRecord).returning('cumulus_id');
```

Comment:
I don't think the `= {}` default assignment is necessary or correct, because the result from a query with `.returning('cumulus_id')` should be an array with the cumulus IDs, not objects

Reply:
True. Let me update to use the correct destructuring.